src/main/java/fi/aalto/cs/apluscourses/intellij/utils/VfsUtil.java

```diff
@@ -4,15 +4,66 @@
 import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.openapi.vfs.VirtualFileVisitor;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import org.jetbrains.annotations.CalledWithReadLock;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public class VfsUtil {
 
   private VfsUtil() {
 
   }
 
+  /**
+   * Recursively looks for a file in the given directory with the given name.
+   *
+   * @param directory The directory from which the search is done.
+   * @param filename  The name of the file that is searched for.
+   * @return A {@link Path} to the found file, or {@code null} if a file isn't found.
+   */
+  @CalledWithReadLock
+  @Nullable
+  public static Path findFileInDirectory(@NotNull Path directory, @NotNull String filename) {
+    VirtualFile virtualFile = com.intellij.openapi.vfs.VfsUtil.findFile(directory, true);
```

Comment:
Nit: `import com.intellij.openapi.vfs.VfsUtil` and `VfsUtilCore` ?

Reply:
Can't do it for `VfsUtil` since our class is named `VfsUtil` as well :smile: but yes for `VfsUtilCore`